lnwallet/channel.go

```diff
@@ -7049,3 +7075,77 @@ func (lc *LightningChannel) unsignedLocalUpdates(remoteMessageIndex,
 
 	return localPeerUpdates
 }
+
+// getCommitmentDeadline finds the deadline for a commitment transaction by
+// extracting the minimum CLTV from its HTLCs. For a local commitment
+// transaction, it uses the lease CLTV value from its incoming HTLCs. For a
+// remote commiment transaction, it uses the least CLTV value from the outgoing
+// HTLCs. If there are no valid HTLCs, the noCommitDeadline is returned to
+// indicate no deadline is needed.
+func (lc *LightningChannel) getCommitmentDeadline(localCommit bool,
+	htlcs []channeldb.HTLC) uint32 {
+
+	// When there are no HTLCs, we have no timeout pressure thus a max
+	// deadline is returned.
+	if len(htlcs) == 0 {
+		return noCommitDeadline
+	}
+
+	lc.channelState.RLock()
+	defer lc.channelState.RUnlock()
+
+	var deadlines = []uint32{}
+
+	for _, htlc := range htlcs {
+		// For local commit tx we use the incoming HTLC's cltv as the
+		// deadline.
+		if localCommit && htlc.Incoming {
+			feeRate := lc.channelState.LocalCommitment.FeePerKw
+			dustLimit := lc.channelState.LocalChanCfg.DustLimit
+
+			// Skip if the HTLC is dust.
+			if htlcIsDust(
+				lc.channelState.ChanType, htlc.Incoming,
+				localCommit, chainfee.SatPerKWeight(feeRate),
+				htlc.Amt.ToSatoshis(), dustLimit,
+			) {
+				continue
+			}
+
+			deadlines = append(deadlines, htlc.RefundTimeout)
```

Comment:
Shouldn't deadlines here be a _relative_ value since that's what's fed in as the _confirmation target_ in the next commit?

Reply:
Yeah my oversight